packages/app/features/checks/utils/checkUtils.ts

```diff
@@ -0,0 +1,136 @@
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts'
+import { assert } from 'app/utils/assert'
+import type { EphemeralKeyPair, ClaimSendCheckPayload } from 'app/features/checks/types'
+import { type Hex, isHex, keccak256, isAddress } from 'viem'
+import type { UserOperation } from 'permissionless'
+import { defaultUserOp } from 'app/utils/useUserOpTransferMutation'
+
+/**
+ * Generates a URL for claiming a /send check.
+ *
+ * The URL encodes information required for /send check retrieval. See {@see ClaimSendCheckPayload } for more information
+ * @param {string} senderId - The sender's /send account id.
+ * @param {EphemeralKeyPair} ephemeralKeyPair - An object containing the ephemeral private key and address.
+ * @returns {string} The generated URL for claiming the check.
+ */
+export const encodeClaimCheckUrl = (
+  senderId: string,
+  ephemeralKeyPair: EphemeralKeyPair
+): string => {
+  const encodedPayload = encodeURIComponent(
+    `${senderId}:${ephemeralKeyPair.ephemeralPrivateKey}:${ephemeralKeyPair.ephemeralAddress}`
+  )
+  return `/checks/claim/${encodedPayload}`
```

Comment:
Add it to the URL fragment. We don't want the browser to send it to the server since we would potentially have knowledge of the key and want to avoid. This would be a security risk for us.

```suggestion
  return `/checks/claim#${encodedPayload}`
```

Reply:
addressed in PR
https://github.com/0xsend/sendapp/pull/617